packages/near-lake-framework/package.json

```diff
@@ -0,0 +1,33 @@
+{
+  "name": "@near-lake-framework/streamer",
```

Comment:
I am not sure I like the idea of deprecating the `near-lake-framework` from NPM.

We can add `near-lake-primitives` and make it a dependency for the framework. We are letting devs install it separately (by `@near-lake/primitives`). I don't insist as I am not aware of the best practices. Just my opinion.

Reply:
Good catch! @morgsmccauley Also, why does it become necessary to publish under the same module name? Can we not roll with near-lake-framework as the package name?

Reply:
After doing some more research i realise that I was misconceived about this.

I thought that in order to publish multiple packages to NPM, you needed a unique account per package. I thought scopes mitigated this by allowing publishing of multiple packages to the same account. But this isn't the case, you _can_ publish multiple packages under the same account, which would make managing `near-lake-framework` and `near-lake-primitives` relatively straight forward.

The benefits of using scoped packages are:
* **No risk of packages names already being taken** - with scopes, only we can publish packages to `@near-lake/` and therefore don't have to worry about names being taken. Without a scope someone could take `near-lake-*` right now.
* **Developers can trust a single organization** - developers can trust that all packages under `@near-lake/` are published by the same organization, i.e. us. Without it they would need to verify the integrity of each individual package, and if they didn't would run the risk of installing something malicious.

Moving from single package to multiple scoped packages is quite common place within the JS ecosystem. For example [`aws-sdk-js` did it from v2 to v3](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/index.html#modularized-packages), and also other popular libraries like [`angular`](https://github.com/angular/angular/tree/main/packages) and [`graphql-tools`](https://the-guild.dev/blog/graphql-tools-v6#new-monorepo-structure). It wouldn't be a surprise for developers to see it happen here.

Whether we also want to go ahead with publishing under a scope depends on how many more packages we expect to publish. If it is just `near-lake-framework` and `near-lake-primitives`, then it might not be worth it, but if we expect more, then a scope would be beneficial.

It's hard to predict the future of this project, so I would recommend going ahead with using a scope. Deprecating `near-lake-framework` is [straight forward](https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions). We can add a deprecation note recommending `@near-lake/framework`, add some more details in the `README.md` here about this decision, and also add a migration doc explaining the required steps for developers. `near-lake-framework` won't go away, people can still use it, but if they need the latest features they'll need to move to `@near-lake/framework`.

Reply:
That sounds good to me. Thank you for the detailed clarification!